include/swupd.h

```diff
@@ -84,6 +84,7 @@ struct file {
 	int last_change;
 	struct update_stat stat;
 
+	unsigned int fd_valid : 1;
```

Comment:
The problem is that there is no explicit initializer function for the struct where one could set -1 reliably in all places; one can only rely on members being cleared (calloc). That's why "fd_valid" != 0 works, but "fd != -1" does not.